clients/drcachesim/tracer/tracer.cpp

```diff
@@ -2902,6 +2974,26 @@ event_thread_exit(void *drcontext)
     if (BUF_PTR(data->seg_base) != NULL) {
         /* This thread was *not* filtered out. */
 
+#ifdef BUILD_PT_TRACER
+        if (op_offline.get_value() && op_enable_kernel_tracing.get_value()) {
+            int recording_sysnum = data->syscall_pt_trace.get_recording_sysnum();
```

Comment:
So the pre syscall event is invoked before the exit event for SYS_exit?

Reply:
Yes.